map_model/src/pathfind/walking.rs

```diff
@@ -419,68 +419,73 @@ pub fn walking_path_to_steps(path: Vec<WalkingNode>, map: &Map) -> Vec<PathStep>
             WalkingNode::LeaveMap(_) => unreachable!(),
         };
 
-        let l1 = r1.must_get_sidewalk(map);
-        let l2 = r2.must_get_sidewalk(map);
-
-        if l1 == l2 {
-            if l1_endpt {
-                steps.push(PathStep::ContraflowLane(l1));
+        if r1 == r2 {
+            if r1_endpt {
+                steps.push(PathStepV2::Contraflow(r1));
             } else {
-                steps.push(PathStep::Lane(l1));
+                steps.push(PathStepV2::Along(r1));
             }
         } else {
-            let i = {
-                let l = map.get_l(l1);
-                if l1_endpt {
-                    l.dst_i
-                } else {
-                    l.src_i
-                }
+            let i = if r1_endpt {
+                r1.dst_i(map)
+            } else {
+                r1.src_i(map)
             };
-            // Could assert the intersection matches (l2, l2_endpt).
-            if let Some(turn) = map.get_turn_between(l1, l2, i) {
-                steps.push(PathStep::Turn(turn));
+            // Could assert the intersection matches (r2, r2_endpt).
+            if map
+                .get_turn_between(r1.must_get_sidewalk(map), r2.must_get_sidewalk(map), i)
+                .is_some()
+            {
+                steps.push(PathStepV2::Turn(MovementID {
+                    from: r1,
+                    to: r2,
+                    parent: i,
+                    crosswalk: true,
+                }));
             } else {
                 println!("walking_path_to_steps has a weird path:");
                 for s in &path {
                     println!("- {:?}", s);
                 }
-                panic!("No turn from {} to {} at {}", l1, l2, i);
+                panic!(
+                    "No turn from {} ({}) to {} ({}) at {}",
+                    r1,
+                    r1.must_get_sidewalk(map),
+                    r2,
+                    r2.must_get_sidewalk(map),
+                    i
+                );
             }
         }
     }
 
     // Don't start or end a path in a turn; sim layer breaks.
-    if let PathStep::Turn(t) = steps[0] {
-        let lane = map.get_l(t.src);
```

Comment:
A bug to watch out for: At first I ported this as `let road = map.get_r(mvmnt.from); if road.src_i == mvmnt.parent...`, but that's not correct! A directed road has a different src/dst intersection than the parent road when the direction is backwards. :)